haystack/nodes/prompt/prompt_node.py

```diff
@@ -905,10 +911,16 @@ def run(
         # to pass results from a pipeline node to any other downstream pipeline node.
         invocation_context = invocation_context or {}
 
+        # prompt_collector is a list that is passed to the prompt node to collect the constructed prompts,
+        # so that can be returned as part of the pipeline's debug output
+        # TODO we'll revisit this approach in the near future and see if we can find a better way collect the prompts
```

Comment:
```suggestion
        # prompt_collector is an empty list, it's passed to the PromptNode that will fill it with the rendered prompts,
        # so that they can be returned by `run()` as part of the pipeline's debug output.
```